src/lib/yang/alarms.lua

```diff
@@ -1,5 +1,50 @@
 module(..., package.seeall)
 
+local data = require('lib.yang.data')
+local util = require('lib.yang.util')
+
+local state = {
+   alarm_inventory = {},
+}
+
+function get_state ()
+   return {
+      alarm_inventory = state.alarm_inventory,
+   }
+end
+
+-- Single point to access alarm type keys.
+alarm_type_keys = {}
+
+function alarm_type_keys:fetch (alarm_type_id, alarm_type_qualifier)
+   self.cache = self.cache or {}
+   local function lookup (alarm_type_id, alarm_type_qualifier)
+      if not self.cache[alarm_type_id] then
+         self.cache[alarm_type_id] = {}
+      end
+      return self.cache[alarm_type_id][alarm_type_qualifier]
+   end
+   assert(alarm_type_id)
+   alarm_type_qualifier = alarm_type_qualifier or ''
+   local key = lookup(alarm_type_id, alarm_type_qualifier)
+   if not key then
+      key = {alarm_type_id=alarm_type_id, alarm_type_qualifier=alarm_type_qualifier}
+      self.cache[alarm_type_id][alarm_type_qualifier] = key
+   end
+   return key
+end
+
+local function load_default_configuration ()
+   local content = require("apps.lwaftr.alarms.lwaftr_default_alarms.inc")
```

Comment:
Would it be possible to make it so that importing an app's module adds appropriate alarms to the global inventory?  Does that make sense at all?